qtazu/widgets/login.py

```diff
@@ -1,11 +1,87 @@
 import os
 import logging
+import re
 
 import gazu
 from Qt import QtWidgets, QtGui, QtCore
 
 log = logging.getLogger(__name__)
 
+PLACEHOLDER = os.path.join(
+    os.path.dirname(os.path.dirname(__file__)), "res", "image", "logo_kitsu.png"
+)
+
+
+class AnimatedLabel(QtWidgets.QLabel):
+    """
+    QLabel with animated background color.
+    """
+
+    def __init__(self):
+        super(AnimatedLabel, self).__init__()
+        self.setStyleSheet(
+            """
+            background-color: #CC4444;
+            color: #F5F5F5;
+            padding: 5px;
+            """
+        )
+        self.setWordWrap(True)
+        self.create_animation()
+
+    def create_animation(self):
+        """
+        Create the animation of the color background.
+        """
+        color_begin = QtGui.QColor("#943434")
+        color_end = QtGui.QColor("#CC4444")
+        byar = QtCore.QByteArray()
+        byar.append("zcolor")
+        self.color_anim = QtCore.QPropertyAnimation(self, byar)
+        self.color_anim.setStartValue(color_begin)
+        self.color_anim.setEndValue(color_end)
+        self.color_anim.setDuration(400)
+
+    def start_animation(self):
+        """
+        Start the animation of the color background.
+        """
+        self.color_anim.stop()
+        self.color_anim.start()
+
+    def parse_style_sheet(self):
+        """
+        Return a list with style sheet instructions.
+        """
+        ss = self.styleSheet()
+        sts = [s.strip() for s in ss.split(';') if len(s.strip())]
+        return sts
+
+    def get_back_color(self):
+        """
+        Get the background color.
+        """
+        return self.palette().color(QtGui.QPalette.Window)
+
+    def set_back_color(self, color):
+        """
+        Set the given color as background color by parsing the style sheet.
+        """
+        sss = self.parse_style_sheet()
+        bg_new = 'background-color: %s' % color.name()
+
+        for k, sty in enumerate(sss):
+            if re.search('\Abackground-color:', sty):
+                sss[k] = bg_new
+                break
+        else:
+            sss.append(bg_new)
+
+        self.setStyleSheet('; '.join(sss))
+
+    # Property to animate : the label background color
+    zcolor = QtCore.Property(QtGui.QColor, get_back_color, set_back_color)
```

Comment:
Where does the label `zcolor` come from? :) I would never assume `zcolor` is the background color. Is that some terminology I'm not familiar with? It's fine with me though. Just wondering.

Reply:
You're right, this naming convention is weird. It makes me think to some naming conventions in computer graphics that are used to manage depth in 3D scenes (z-depth, z-buffer, ...). But in our case I doubt we want to refer to that. I change it to background_color.